tracer/test/Datadog.Trace.Tests/Logging/DatadogLoggingTests.cs

```diff
@@ -205,6 +209,40 @@ public void ErrorsDuringLogging_DontBubbleUp()
             mockLogger.Verify();
         }
 
+        [Fact]
+        public void DuringStartup_OldLogFilesGetDeleted()
+        {
+            var tempLogsDir = Path.Combine(Path.GetTempPath(), "Datadog .NET Tracer\\logs");
+            Directory.CreateDirectory(tempLogsDir);
+
+            // Creating log files that match expected formats
+            var logPath = Path.Combine(tempLogsDir, "DD-DotNet-Profiler-Native-1.log");
+            File.Create(logPath).Dispose();
+            File.SetLastWriteTime(logPath, DateTime.Now.AddDays(-36));
+
+            File.Create(Path.Combine(tempLogsDir, "dotnet-tracer-managed-2.log")).Dispose();
+            File.Create(Path.Combine(tempLogsDir, "dotnet-tracer-native-3.log")).Dispose();
+
+            for (int i = 0; i < 3; i++)
+            {
+                // Adding random files that don't match the pattern
+                var mockFilePath = Path.Combine(tempLogsDir, $"random-file-{i}.txt");
+                File.Create(mockFilePath).Dispose();
+                File.SetLastWriteTime(mockFilePath, DateTime.Now.AddDays(-31 * i));
+            }
+
+            // Running method to delete the old files
+            DatadogLogging.CleanLogFiles(tempLogsDir);
+
+            // Asserting that the correct amount of files are left
+            Directory.EnumerateFiles(tempLogsDir, "DD-DotNet-Profiler-Native-*").Count().Should().Be(0);
```

Comment:
I liked before that you were running the assertions _after_ doing the test cleanup (`Directory.Delete`). This should make it easier because if one of the asserts fails, it will fail (and throw an exception) after running the `Directory.Delete` line

```suggestion
            using var scope = new AssertionScope();
            Directory.EnumerateFiles(tempLogsDir, "DD-DotNet-Profiler-Native-*").Count().Should().Be(0);
```

Reply:
That makes sense, I updated the assertions now, let me know what you think about them since I wasn't sure if you had a different idea based on the first comment of your previous review:

> Maybe we could also calculate it beforehand as well, so we could write some assertion like deletedFileCount.Should().Be(1);

Reply:
The changes look fine to me 👍🏼